### PR TITLE
Add trailing slashes to URL routes (#895)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR seeks to establish a consistent approach for defining URL routes by adding trailing forward slashes (where possible) to existing routes. We discovered that missing slashes can cause problems during testing of the Canvas integration (see #893). Trailing forward slashes are included in examples in the [Django documentation](https://docs.djangoproject.com/en/2.2/ref/urls/). With these added, the [`APPEND_SLASH` setting](https://docs.djangoproject.com/en/3.0/ref/settings/#append-slash) (part of the `CommonMiddleware`, it defaults to `True`) should redirect users if they try to visit one of these routes without the trailing slash. However, redirects for `PUT` or `POST` requests will not retain data, so I needed to change the `useSetUserSetting` hook to use the trailing slash (Alternatively, we could not use a trailing slash here, but it seems nice to be consistent). The PR aims to resolve issue #895.

### CODE REVIEW

**Comment by pushyamig** [comment]
Looks good, just need to resolve the conflicts before merging

File: dashboard/urls.py [hunk 1]
```diff
@@ -33,8 +33,9 @@
...
     path('', views.get_home_template, name = 'home'),
     path('status/', include('watchman.urls')),
-    path('status/bare_status', watchman.views.bare_status),
-    path('admin', admin.site.urls),
+    path('status/bare_status/', watchman.views.bare_status),
+
+    path('admin/', admin.site.urls),
 
     # graphql access
     path('graphql', DashboardGraphQLView.as_view(graphiql=True)),
```
Comment (ssciolla) [inline dashboard/urls.py]: I tried adding a `/` here but that seemed to break the GraphQL implementation. Part of the problem seems to be that `PUT` calls will lose data when the slash is appended and a redirect happens. I was hoping there would be a way to add `/` to calls in the front-end -- like we did with `userSetUserSetting` -- but I wasn't able to find anything easily. @justin0022, any suggestions? I'm fine keeping it without the `/` though, it'd be nice to be consistent.

**Comment by ssciolla** [comment]
I still need to deploy to the test environment to ensure the SAML routes work. I'll try to do that today or later this week.

File: dashboard/urls.py (hunk 1 above)
Comment (justin0022) [reply dashboard/urls.py]: Hey sorry @ssciolla I'll look into this. Didn't see this message till now.

**Comment by ssciolla** [comment]
Okay, I hit a hiccup when testing the SAML integration. Logging in worked, but I encountered a strange error page when trying to access `/accounts/logout/` (something like `'Connection aborted.', OSError("(104, 'ECONNRESET')"`. For now, I have rolled back the change to the override in the SAML block (`accounts/logout`), which seems to have fixed this issue.

When talking with @jonespm about this, we also noticed that the FE's `logout` button was not using the URL from `settings.py`, despite it being passed as part of `myla_globals`. I made a few changes to the FE (mainly moving the `loginURL` and `logoutURL` into `user`) to fix this.

File: dashboard/urls.py (hunk 1 above)
Comment (justin0022) [reply dashboard/urls.py]: I'm not sure what the best way to address this is

**Comment by ssciolla** [comment]
@pushyamig, @zqian, and @jonespm, do you think it's okay to merge this? Note that I've elected to leave the SAML and GraphQL routes alone for the moment.

File: dashboard/urls.py (hunk 1 above)
Comment (ssciolla) [reply dashboard/urls.py]: K, let's leave it alone for now.

**Comment by ssciolla** [comment]
> Yeah, could you just add a comment to the code by these two endpoints you didn't change so if someone comes in and wonders why we didn't do slashes here why we left them as is? Then I think it's good.

Okay @jonespm, let me know if the latest commit works for you.

**Comment by jonespm** [comment]
Looks great thanks @ssciolla

**Comment by ssciolla** [comment]
FYI, I just added another comment to `settings.py` explaining why `LOGOUT_URL` doesn't have the trailing slash. Merging now.

**Pull request merged**